### PR TITLE
Added python type preservation in parameter passing

### PR DESCRIPTION
**Pull request by MSeal**
Passing a json object will give you a dict in your Python notebook instead of a string. Same for lists, numbers, and booleans. This makes passing complex parameters very nice for consumption in the notebook.

Broken out from https://github.com/nteract/papermill/pull/88

### CODE REVIEW

File: papermill/tests/test_execute.py [hunk 1]
```diff
@@ -9,12 +9,46 @@
...
 
 from ..api import read_notebook
-from ..execute import execute_notebook, log_outputs
+from ..execute import execute_notebook, log_outputs, _form_escaped_python_value
 from ..exceptions import PapermillExecutionError
 from . import get_notebook_path, RedirectOutput
 
 python_2 = sys.version_info[0] == 2
 
+class TestPythonEscapers(unittest.TestCase):
```
Comment (willingc) [inline papermill/tests/test_execute.py]: I believe you could refactor this to one test case with parameters using pytest.

File: papermill/execute.py [hunk 2]
```diff
@@ -240,6 +240,20 @@ def _form_escaped_value(str_val):
 # Registry for functions that build parameter assignment code.
 _parameter_code_builders = {}
 
+def _form_escaped_python_value(val):
```
Comment (willingc) [inline papermill/execute.py]: Let's consider renaming to `_translate_type_python`, `_translate_str_python`, `_translate_list_python`.... See https://github.com/nteract/papermill/pull/93#discussion_r161518590 for rationale.

File: papermill/tests/test_execute.py (hunk 1 above)
Comment (MSeal) [reply papermill/tests/test_execute.py]: Hmm true, seems odd to have lots of mixed pytest and unittest components. Are we planning on converting everything over to pytest or just using whichever is cleaner for particular tests? I'll convert this one, just curious on overall direction you see.

Comment (betatim) [reply papermill/tests/test_execute.py]: My vote is pytest all the way.

Comment (willingc) [reply papermill/tests/test_execute.py]: @MSeal The preferred approach for new tests would be to use pytest as it is far less verbose and maintainable than unittest. The pytest runner will collect tests from either so we don't need to convert from unittest for existing tests. A good project with very clean tests is `requests`.

File: papermill/execute.py [hunk 3]
```diff
@@ -229,17 +229,52 @@ def _find_parameters_index(nb):
         raise PapermillException("Multiple parameters tags found")
     return parameters_indices[0]
 
-def _form_escaped_value(str_val):
+def _translate_escaped_str(str_val):
+    '''Reusable by most interpreters'''
```
Comment (willingc) [inline papermill/execute.py]: Style tip: use triple double quotes for docstrings `"""`

Comment (MSeal) [reply papermill/execute.py]: Opps, yep, fixing.

**Pull request merged**